_posts/2020-06-28-openfaas-loki-into.md

```diff
@@ -0,0 +1,235 @@
+        (____)
+
+         (__)
+         (oo)
+  /-------\/
+ / |     ||
+*  ||----||
+   ~~    ~~
+   holy cow
+```
+
+We can see the logs powered by Kubernetes using
+
+```bash
+faas-cli logs cows
+WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.
+Handling connection for 8080
+2020-06-27T16:37:13Z 2020/06/27 16:37:13 Version: 0.13.0	SHA: fa93655d90d1518b04e7cfca7d7548d7d133a34e
+2020-06-27T16:37:13Z 2020/06/27 16:37:13 Read/write timeout: 5s, 5s. Port: 8080
+2020-06-27T16:37:13Z 2020/06/27 16:37:13 Writing lock-file to: /tmp/.lock
+2020-06-27T16:37:13Z 2020/06/27 16:37:13 Metrics server. Port: 8081
+2020-06-27T16:37:17Z 2020/06/27 16:37:17 Forking fprocess.
+2020-06-27T16:37:17Z 2020/06/27 16:37:17 Wrote 119 Bytes - Duration: 0.130287 seconds
+```
+
+We will see a slightly different format when we switch to Loki later.
+
+### Install Loki and Grafana
+
+With `arkade`, we need this single line to install both Grafana _and_ Loki:
+
+```bash
+arkade install loki --grafana
+```
+
+Wati for Loki and Grafana to be ready:
+
+```bash
+kubectl rollout status deployment loki-stack-grafana
+```
+
+Now get your admin password and forward the Grafana port
+```bash
+kubectl get secret loki-stack-grafana -o jsonpath="{.data.admin-password}" | base64 --decode ; echo
+kubectl port-forward service/loki-stack-grafana 3000:80 &
+```
+
+Then open `http://localhost:3000` and login using username `admin`.
+
+The cool thing about Loki is that it will automatically start collecting logs from _all_ of the Pods it can sees. Importantly, this means you can see the logs for all of the OpenFaaS components, like the Gateway.
+
+This [link](http://localhost:3000/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Loki%22,%7B%22expr%22:%22%7Bapp%3D%5C%22gateway%5C%22%7D%22%7D,%7B%22mode%22:%22Logs%22%7D,%7B%22ui%22:%5Btrue,true,true,%22none%22%5D%7D%5D) will show you the logs (over the last hour) for the Gateway we just installed!
+
+Loki, when installed in Kubernetes, will automatically create a search index for each of the lables on the Pods it sees. Important, this means the Loki query for the logs of an OpenFaaS function looks like `{faas_function="<name>"}` because OpenFaaS always adds the function name as the label `faas_function`.  This [link](http://localhost:3000/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Loki%22,%7B%22expr%22:%22%7Bfaas_function%3D%5C%22cows%5C%22%7D%22%7D,%7B%22mode%22:%22Logs%22%7D,%7B%22ui%22:%5Btrue,true,true,%22none%22%5D%7D%5D) will show the logs for the Cows function we installed and invoked.
+
+### Installing `openfaas-loki`
+
+To combine OpenFaaS and Loki, we need to install `openfaas-loki`:
+
+```bash
+arkade install openfaas-loki
+```
+
+We then need to configure the OpenFaaS installation, we can do this with `arkade` using the `--set` flag
+
+```bash
+arkade install openfaas --basic-auth=false --set gateway.logsProviderURL=http://openfaas-loki.default:9191/
```

Comment:
Not sure `.default` is correct after the latest arkade release.

Reply:
i used arkade built from master